bot.py

```diff
@@ -574,30 +730,29 @@ async def play(self, message, command_content, playnext=False):
         if not voice_client:
             return
 
+        # Empty play command
         if not command_content:
-            # No search term/url
-            if self.voice_client.is_paused():
-                await self.resume(message, command_content)
-            elif not self.voice_client.is_playing() and len(self.media_deque) != 0:
-                await self.resume(message, command_content)
-            elif self.voice_client.is_playing():
-                logging.info("User %s tried 'play' with no search term", message.author)
-                await message.channel.send(
-                    ":unamused: Please enter something to search!"
-                )
-            else:
-                logging.info("User %s tried 'play' on empty queue", message.author)
-                await message.channel.send(
-                    ":unamused: Queue is empty - please enter something to search!"
-                )
+            self.play_empty(message, command_content)
             return
 
-        if re.search(self.playlist_regex, command_content):
+        # Playlist/album
+        if (
+            re.search(self.playlist_regex, command_content)
+            or re.search(self.spotify_album_regex, command_content)
+            or re.search(self.spotify_playlist_regex, command_content)
+        ):
             await self.playlist(message, command_content)
             return
 
-        media = self.get_media(message, command_content)
+        # Single track/url/search term
+        media = None
+        if re.search(self.spotify_track_regex, command_content):
+            media = self._get_spotify_tracks(command_content)[0]
```

Comment:
I think reusing the SpotifyList class here is quite clever, but imo it makes this much less  readable than `media = self.get_media()`. I think it might be a good idea to add a comment here briefly explaining what is going on.

Reply:
Comment added.